### PR TITLE
Feat/search

### PR DESCRIPTION
**Pull request by sheilaazhara**
### Overview
- Added types of each pokemon
- Background color based on their type
- Search function based on pokemon name

### Preview
Screenshot (616)
Screenshot (615)

### CODE REVIEW

File: src/assets/css/main.css [hunk 1]
```diff
@@ -1,3 +1,43 @@
...
+.normal {background:rgb(67, 67, 67) !important;}
+.psychic {background:rgb(195, 0, 255) !important;}
+.bug {background:rgb(52, 87, 6) !important;}
+.dark {background:rgb(43, 43, 43) !important;}
+.steel {background:rgb(116, 116, 116) !important;}
+.fairy {background:rgb(186, 112, 176) !important;}
+.electric {background:rgb(147, 127, 11) !important;}
+.rock {background:rgb(88, 95, 100) !important;}
+.ground {background:rgb(92, 70, 70) !important;}
+.unknown {background:rgb(138, 138, 138) !important;}
```
Comment (Ibwedagama) [inline src/assets/css/main.css:43]: minor/discussion:

hello, @sheilaazhar thanks for your hard work 👍
alternative to this approach we can extend classes from tailwind itself using `tailwind.config.js` but your method is absolutely fine

Here are just some tips and alternative ways to extend tailwind default styling:

https://tailwindcss.com/docs/background-color
https://tailwindcss.com/docs/text-color

**Review by Ibwedagama — APPROVED** [review]

File: src/assets/css/main.css (hunk 1 above)
Comment (sheilaazhara) [reply src/assets/css/main.css:43]: Oh I didn't know that. Thankyou for the tips @Ibwedagama

File: src/components/CardPokemon/index.vue [hunk 2]
```diff
@@ -1,26 +1,43 @@
...
-      <span class="text-xs text-black">#{{ pokemon.id }}</span>
+  <article :class="('relative pt-5 pb-3 rounded-lg shadow flex flex-col items-center transition ease ring-2 ring-transparent hover:ring-emerald-500 hover:ring-offset-2 bg-opacity-20 bg-'+types[0].type.name)">
+    <div class="absolute leading-none top-[2px] left-[2px] rounded-tl-lg rounded py-1 px-2 flex items-center bg-gray-100 bg-opacity-60">
+      <span class="text-xs text-black">#{{ id }}</span>
     </div>
     <img
-      :src="imageUrl + pokemon.id + '.png'"
-      :alt="pokemon.name + ' image'"
+      :src="imageUrl + id + '.png'"
+      :alt="name + ' image'"
```
Comment (naufalihsank) [inline src/components/CardPokemon/index.vue]: improvement suggestion

we usually put this kind of param/input into some `methods` or `computed` to make it more maintainable

so it'll be like this
```vue
<img
  :src="imageSource"
  :alt="alt"
  height="140"
  width="140"
  class="h-auto w-36"
>
```
```js
computed: {
	imageSource() {
		return `${this.imageUrl}${this.id}.png`
	},
	alt() {
		return `${this.name} image`
	}
}
```

**Review by doohanas — APPROVED** [review]

**Review by naufalihsank — APPROVED** [review]

File: src/components/CardPokemon/index.vue (hunk 2 above)
Comment (sheilaazhara) [reply src/components/CardPokemon/index.vue]: Oh I see, thank you @naufalihsank for the suggestion. I've fixed it [here](https://github.com/sheilaazhar/fe-onboarding-mini-task/pull/5/commits/41400af63db39d5a506de460972d07094d11287c)

**Pull request merged**